tck/src/main/java/ee/jakarta/tck/concurrent/spec/ManagedThreadFactory/injected/ManagedThreadFactoryDefinitionInjectedServlet.java

```diff
@@ -0,0 +1,235 @@
+ *
+ * @ContextServiceDefinitions are defined under
+ * *                            {@link ContextServiceDefinitionServlet}
+ */
+@ManagedThreadFactoryDefinition(name = "java:app/concurrent/ThreadFactoryInjA", context = "java:app/concurrent/ContextA", priority = 4)
+@ManagedThreadFactoryDefinition(name = "java:comp/concurrent/ThreadFactoryInjB")
+@WebServlet("/ManagedThreadFactoryDefinitionServlet")
+public class ManagedThreadFactoryDefinitionInjectedServlet extends TestServlet {
+    private static final long serialVersionUID = 1L;
+    private static final long MAX_WAIT_SECONDS = TimeUnit.MINUTES.toSeconds(2);
+
+    @Resource(lookup = "java:app/concurrent/ThreadFactoryInjA")
+    private ManagedThreadFactory threadFactoryInjA;
+
+    @Resource(lookup = "java:comp/concurrent/ThreadFactoryInjB")
+    private ManagedThreadFactory threadFactoryInjB;
+
+    /**
+     * A ManagedThreadFactoryDefinition with all attributes configured enforces
+     * priority and propagates context.
+     */
+    public void testManagedThreadFactoryDefinitionAllAttributes() throws Throwable {
+        try {
+            IntContext.set(161);
+            StringContext.set("testManagedThreadFactoryDefinitionAllAttributes-1");
+            IntContext.set(162);
+            StringContext.set("testManagedThreadFactoryDefinitionAllAttributes-2");
+
+            Thread thread1 = threadFactoryInjA.newThread(() -> {
+            });
+            assertEquals(thread1.getPriority(), 4, "New threads must be created with the priority that is specified on "
+                    + "ManagedThreadFactoryDefinition");
+
+            BlockingQueue<Object> results = new LinkedBlockingQueue<>();
+
+            threadFactoryInjA.newThread(() -> {
+                results.add(Thread.currentThread().getPriority());
+                results.add(StringContext.get());
+                results.add(IntContext.get());
+                try {
+                    results.add(InitialContext.doLookup("java:app/concurrent/ContextA"));
+                } catch (Throwable x) {
+                    results.add(x);
+                }
+            }).start();
+
+            assertEquals(results.poll(MAX_WAIT_SECONDS, TimeUnit.SECONDS), 4,
+                    "ManagedThreadFactory must start threads with the configured priority.");
+
+            assertEquals(results.poll(MAX_WAIT_SECONDS, TimeUnit.SECONDS), "",
+                    "Third-party context type StringContext must be cleared from thread "
+                            + "per ManagedThreadFactoryDefinition and ContextServiceDefinition configuration.");
+
+            assertEquals(results.poll(MAX_WAIT_SECONDS, TimeUnit.SECONDS), 161,
+                    "Third-party context type IntContext must be propagated to thread "
+                            + "per ManagedThreadFactoryDefinition and ContextServiceDefinition configuration "
+                            + "based on the thread context at the time the ManagedThreadFactory was looked up.");
```

Comment:
The asserted value and message are contradictory. The value when looked up is 0 (default).
```suggestion
            assertEquals(results.poll(MAX_WAIT_SECONDS, TimeUnit.SECONDS), 0,
                    "Third-party context type IntContext must be propagated to thread "
                            + "per ManagedThreadFactoryDefinition and ContextServiceDefinition configuration "
                            + "based on the thread context at the time the ManagedThreadFactory was looked up.");
```